homeassistant/components/automation/__init__.py

```diff
@@ -119,6 +119,7 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+IF_ACTION = Callable[[Optional[Mapping[str, Any]]], bool]
```

Comment:
@cdce8p I need your help on this one as the `Callable` gets an extra attribute in:
https://github.com/home-assistant/core/blob/52ea9998bbc47f53f493d22e747e43c3c3bf3deb/homeassistant/components/automation/__init__.py#L783-L785

Reply:
You can use a Protocol type for that.
```py
class IF_ACTION(Protocol):
    config: list[dict[str, Any]]  # or another type :)

    def __call__(self, variables: Mapping[str, Any] | None = None) -> bool:
        ...
```

Later in `async def _async_process_if` add `if_action: IF_ACTION  # type: ignore[no-redef]` before `def if_action`.